code/components/discord/src/DiscordPresence.cpp

```diff
@@ -193,6 +202,24 @@ static InitFunction initFunction([]()
 
 	OnGameFrame.Connect([]()
 	{
+		if (!g_richPresenceEnabled)
+		{
+			if (g_wasRichPresencePreviouslyEnabled)
+			{
+				g_wasRichPresencePreviouslyEnabled = false;
+				Discord_Shutdown();
+			}
+			return;
+		}
+
+		if (!g_wasRichPresencePreviouslyEnabled)
+		{
+			g_wasRichPresencePreviouslyEnabled = true;
+			DiscordEventHandlers handlers;
+			memset(&handlers, 0, sizeof(handlers));
+			Discord_Initialize(g_discordAppId.c_str(), &handlers, 1, nullptr);
```

Comment:
Something I didn't think about before, there should be a `g_richPresenceChanged = true;` after all of this so `UpdatePresence()` gets ran.

Reply:
Makes sense to get an instant update/re-activation. Resolved.